### ISSUE TITLE
Bag priority

### ISSUE THREAD

**Issue by YaibaToKen** [aura-project/aura#366]
As far as I'm aware, in NA, when one has an Ore Bag in the inventory, with sufficient space, and picks up some ores, they go into the Ore Bag instead of the regular inventory. Afaik this doesnt happen in Aura, with any bag at all.. as far as I've tested at least.

Disregard if this is intentional though.

**Comment by exectails** [comment]
I wouldn't say it's intentional, it's simply a feature that hasn't been implemented yet. I've kinda been glossing over it, since it's not needed for G1. I'll put it on the backburner, maybe someone feels like implementing it. It's not too complicated in theory, when looking for a pocket to add to we just have to treat some of them with a higher priority, basically checking those first.

**Comment by YaibaToKen** [comment]
So.. basically we search for the bag with the highest priority, check if it has room, place it there, or search for the next highest priority and so on? I could try looking into that

**Comment by exectails** [comment]
Yes, basically. The only question is how exactly to do that search.

**Comment by YaibaToKen** [comment]
Sorted list based on depth and what the bag can carry/already carries. So if you have a potion bag and the main inventory, the potion bag gets priority for potions, and if you have 2 potion bags, you'd pick the one that already has some of those potions. If neither of them have that item type, you pick the one that's closest to the bottom as I think that is the order mabi takes when you use items from your inventory. Does that seem understandable and simple enough? I can try and write an algorithm for it, probably just pseudocode but still better than nothing.

**Comment by exectails** [comment]
If we assume that adding to existing stacks is the default, no matter where they are, all we need is a search for special bags in Add I guess.

**Comment by YaibaToKen** [comment]
Sounds like a solution :) do we have a way of knowing what goes inside a specific bag? or in what bags can you carry a particular item?

**Comment by exectails** [comment]
No idea, I haven't looked into that yet *at all*.

**Comment by YaibaToKen** [comment]
What should happen if there are 2 incomplete stacks of the same item, one in the special bag and the other in a regular bag? I'd say the special one should take priority, though that would mean you'd have to iterate through every single inventory slot available, just to check if stack exists, stack isnt full and bag is special... Though I guess you already do something similar checking if stack exists and isnt full so adding a check to see if the bag is special shouldnt be hard in itself. Should I close this issue? Or do you prefer it being left open for now?

**Comment by exectails** [comment]
It's something we do have to implement at some point, so it can stay open.

I don't know how officials operate in the case you're describing, but that's what we want to emulate. We basically have to test it.

**Comment by YaibaToKen** [comment]
Alright, also, regarding what the bag can have in it and what not, was fiddling around on NA and remembered the bag tags :)

**Comment by exectails** [comment]
I just realized this functionality might be much more fundamental than we thought. The inventory consists of multiple "pockets". When an item is added to the inventory, it's added to the pocket best suited for it, which we associated with certain bags that are made for certain types here, but there's another case where this has happened from the very beginning: equipment.

I'm pretty sure we don't have that right now, but when you get equipment, from an NPC, or by picking it up, it doesn't go into your inventory, but you equip it instead. It's the the same concept, and probably uses the same system, the equip pockets have the highest priority for their respective equip type.
